### PR TITLE
559107 - Reprocessor - Business plan - Facade

### PR DESCRIPTION
**Pull request by gurpal-kahlon**

### CODE REVIEW

File: src/EPR.RegulatorService.Facade.Core/Services/ReprocessorExporter/Registrations/ReprocessorExporterService.cs [hunk 1]
```diff
@@ -287,4 +286,35 @@ public async Task<bool> SaveRegistrationTaskQueryNotes(Guid id, Guid userId, Que
...
+    public async Task<AccreditationBusinessPlanDto> GetBusinessPlanByAccreditationId(Guid id)
+    {
+        var businessplanInfo =  await reprocessorExporterServiceClient.GetBusinessPlanByAccreditationId(id);
+        var organisationDetails = await accountServiceClient.GetOrganisationDetailsById(businessplanInfo.OrganisationId);
+
+        return new AccreditationBusinessPlanDto
+        {
+            AccreditationId = businessplanInfo.AccreditationId,
+            OrganisationId = businessplanInfo.OrganisationId,
+            OrganisationName = organisationDetails.OrganisationName,
```
Comment (paulshimmen) [inline src/EPR.RegulatorService.Facade.Core/Services/ReprocessorExporter/Registrations/ReprocessorExporterService.cs:299]: I think it is worth adding the null check here to align the the rest of the service. ie OrganisationName = organisationDetails?.OrganisationName,

File: src/EPR.RegulatorService.Facade.UnitTests/Services/ReprocessorExporter/Registrations/ReprocessorExporterServiceTests.cs [hunk 2]
```diff
@@ -1042,4 +1043,49 @@ public async Task SaveRegistrationTaskQueryNotes_ShouldReturnExpectedResult()
...
+        _mockReprocessorExporterServiceClient
+            .Setup(client => client.GetBusinessPlanByAccreditationId(id))
+            .ThrowsAsync(new Exception("Service unavailable"));
+
+        _service = new ReprocessorExporterService(_mockReprocessorExporterServiceClient.Object, _mockAccountsServiceClient.Object, _mockPaymentServiceClient.Object);
+
+        await FluentActions.Invoking(() => _service.GetBusinessPlanByAccreditationId(id))
+                            .Should().ThrowAsync<Exception>()
+                            .WithMessage("Service unavailable");
+    }
```
Comment (paulshimmen) [inline src/EPR.RegulatorService.Facade.UnitTests/Services/ReprocessorExporter/Registrations/ReprocessorExporterServiceTests.cs:1090]: I think we should add an additional test here to cover the potential for the org name being null. ie GetBusinessPlanByAccreditationId_ShouldReturnDto_WhenOrganisationIsNull

**Review by paulshimmen — CHANGES_REQUESTED** [review]

File: src/EPR.RegulatorService.Facade.UnitTests/Services/ReprocessorExporter/Registrations/ReprocessorExporterServiceTests.cs (hunk 2 above)
Comment (gurpal-kahlon) [reply src/EPR.RegulatorService.Facade.UnitTests/Services/ReprocessorExporter/Registrations/ReprocessorExporterServiceTests.cs:1090]: As discussed - leaving as is because this would highlight missing org details which is a serious data issue

**Review by paulshimmen — APPROVED** [review]
